pip/qsharp/_qsharp.py

```diff
@@ -441,16 +441,16 @@ def callback(output: Output) -> None:
         return get_interpreter().invoke(callable, args, callback)
 
     # Each callable is annotated so that we know it is auto-generated and can be removed on a re-init of the interpreter.
-    _callable.__qs_gen = True
+    _callable.__global_callable = callable
 
     # Add the callable to the module.
     module.__setattr__(callable_name, _callable)
 
 
 def run(
-    entry_expr: str,
+    entry_expr: Union[str, Callable],
```

Comment:
Maybe this makes more sense. Or is it the 'back compat' concern which is causing us to just expand params on the other APIs rather than union them?

Reply:
This is where I'm using more Pythonic patterns to expand what types the user-facing API can accept without breaking existing code or extending the number of parameters. I can't do that in __native.pyi the same way because those are interop wrappers around the native Rust code, so if I wanted to have the equivalent of a `Union` I'd need to introduce a Rust struct and then reflect that via pyO3 as a class into the Python wrapper, which seemed like overkill for what is essentially an internal API.